### PR TITLE
#179 Add initialize at build time flag

### PR DESCRIPTION
**Pull request by jperedadnr**

### CODE REVIEW

File: src/main/java/com/gluonhq/substrate/model/InternalProjectConfiguration.java [hunk 1]
```diff
@@ -307,16 +308,16 @@ public boolean isUseLLVM() {
...
-    public List<String> getDelayInitList() {
-        return delayInitList;
+    public List<String> getInitBuildTimeList() {
+        return initBuildTimeList;
     }
 
     /**
      * Sets additional lists
-     * @param delayInitList a list of classes that will be added to the default delayed list
+     * @param initBuildTimeList a list of classes that will be added to the default delayed list
```
Comment (johanvos) [inline src/main/java/com/gluonhq/substrate/model/InternalProjectConfiguration.java]: that will be added to the initialize-at-buildtime list (as opposed to delayed)

File: src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java [hunk 2]
```diff
@@ -106,6 +106,10 @@ public boolean compile(String cp) throws IOException, InterruptedException {
         useGlisten = GlistenResolver.useGlisten(cp);
         String nativeImage = getNativeImagePath();
         ProcessBuilder compileBuilder = new ProcessBuilder(nativeImage);
+        List<String> buildTimeList = getInitializeAtBuildTimeList(useGlisten);
```
Comment (johanvos) [inline src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java]: This is one for @eugener : do we want to have this boolean inside the method, or on the configuration?

File: src/main/java/com/gluonhq/substrate/model/InternalProjectConfiguration.java (hunk 1 above)
Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/model/InternalProjectConfiguration.java]: Fixed

File: src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java (hunk 2 above)
Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java]: Agree, we'll need a better way to handle the different "uses", but I'd say we can do that in a follow up PR?

Comment (eugener) [reply src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java]: Agree with @jperedadnr... we should try to redesign the resolvers using "strategy" pattern. But that can be done in a later PR - no sense of holding this one.

File: src/main/java/com/gluonhq/substrate/util/Strings.java [hunk 3]
```diff
@@ -73,4 +76,12 @@ public static String substitute(String template, Map<String,String> context) {
         return substitute(template, context::get);
     }
 
+    /**
+     * Converts a string of comma separated strings into an array of strings
+     * @param s a string that contains comma separated strings
+     * @return a list of strings
+     */
+    public static List<String> splitString(String s) {
```
Comment (eugener) [inline src/main/java/com/gluonhq/substrate/util/Strings.java]: Good decision! I totally forgot about this one while creating the Strings class.

**Pull request merged**